tests/transport_test.go

```diff
@@ -0,0 +1,118 @@
+package tests
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+
+	"k8s.io/api/core/v1"
+)
+
+var _ = Describe("Transport Tests", func() {
+
+	const (
+		annEndpoint = "cdi.kubevirt.io/storage.import.endpoint"
+		annSecret   = "cdi.kubevirt.io/storage.import.secretName"
+
+		fileHostName   = "cdi-file-host" // deployment and service name
+		fileHostNs     = "kube-system"   // deployment and service namespace
+		fileHostUname  = "admin"         // http and s3 username
+		fileHostPasswd = "password"      // http and s3 password
+
+		httpPortName = "http"
+		s3PortName   = "s3"
+
+		targetFile = "tinyCore.iso"
+		bucket     = "images"
+	)
+
+	var (
+		httpEp, s3Ep, ns string
+		fileHostService  *v1.Service // persistent file host service, routes to http and s3 ports
+	)
+
+	f, err := framework.NewFramework("", framework.Config{false})
+	if err != nil {
+		Fail(errors.Wrap(err, "Failed to create test framework").Error())
+	}
+
+	c, err := f.GetKubeClient()
```

Comment:
f.K8sClient will get you the same thing without creating a new instance of the client.

Reply:
@awels I tried that initially but get nil ptr errors making `CoreV1()` calls.

Reply:
Ahh yes, the K8sClient is created in the BeforeEach block, and at this point that hasn't run yet. We should probably move that into the NewFramework function.

Reply:
+1. Jon you can add that to this pr or do a quick separate pr to fix this.

Reply:
PR #401 moves the client creation into NewFramework.